crates/matrix-sdk/src/widget/permissions.rs

```diff
@@ -23,4 +23,11 @@ pub struct Permissions {
     pub read: Vec<EventFilter>,
     /// Types of the messages that a widget wants to be able to send.
     pub send: Vec<EventFilter>,
+    /// If this permission is requested by the widget the client should not
+    /// provide the option to pop the widget into its own window.
+    ///
+    /// If the widget does not work without the widget api (e.g. it relies on
+    /// asking for matrix room messages) it will(/has to) request this
+    /// permission.
```

Comment:
Still no clue how popping into its own window is related to client permissions, but this doc comment is definitely better than before. Can you copy it over to the FFI crate?

Reply:
I will add some background to this here. Maybe you have an idea what is missing in the doc to make it more clear.
A client (only EW atm) can provide a button to pop out the IFrame into its own window so a widget can be used on a seperate screen for instance. This breaks the widget api communication over postmessage. Some widgets are not usable then (e.g. EC). For a widget to communicate that it wont work anymore if the client decides to pop it out into its own window a capability was introduced. (Its not the most straight forward capability but it is something the widget need from the client to be fully operational. (in this case its inverse. its something the client is NOT allowed to do to make the widget operational). If the widget "requests the capability" (tells the client it will only work if it can use postmessages/ it requires the client) it asks the client for the`io.element.requires_client` capability. Form this point on the client should hide/disable all ux flows that would allow to pop out the widget into its own window. EW just hides the according button.

Reply:
By "its own window", do you mean opening the widget in the system's default browser?

Reply:
yes exacty